### PR TITLE
updating auth

### PR DESCRIPTION
**Pull request by will-iamalpine**
parsing the info & adding images/explanations

### CODE REVIEW

File: .cloud/.azure/workspace.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "<your-workspace-name>",
-    "resource_group": "<your-resource-group-name>",
+    "name": "wibuchan-workspace",
+    "resource_group": "wibuchan",
```
Comment (marvinbuss) [inline .cloud/.azure/workspace.json]: @buchananwp We should not put your workspace details in the template. Can you please change it back to the placeholders?

File: .autogit/autogit.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "updateInterval": 1800,
+  "logging": true,
+  "silent": true
+}
```
Comment (marvinbuss) [inline .autogit/autogit.json]: What is this file for? I think we should not include this in the template repository. Would be great to understand how this helps the user.

File: README.md [hunk 3]
```diff
@@ -51,31 +71,57 @@ This will generate the following JSON output:
...
+Where required, modify the environment yaml so that the training and deployment environments will have the correct packages installed in the conda environment for your training and deployment.
+
+Upon pushing the changes, actions will kick off your training and deployment run. 
+
+Check the actions tab to view if your actions have successfully run.
 
-Comment lines 39 to 55 in your <a href="/.github/workflows/train_deploy.yml">`"/.github/workflows/train_deploy.yml"` file</a> if you only want to train the model. Uncomment line 7 to 8, if you only want to kick off the workflow when pushing changes to the `"/code/"` file.
+OPTIONAL: Comment lines 39 to 55 in your <a href="/.github/workflows/train_deploy.yml">`"/.github/workflows/train_deploy.yml"` file</a> if you only want to train the model. Uncomment line 7 to 8, if you only want to kick off the workflow when pushing changes to the `"/code/"` file.
 
 ### 6. Viewing your AML resources and runs
```
Comment (marvinbuss) [inline README.md]: Thanks for the details. That helps users a lot. Good job!
This is great. Thanks!

**Review by marvinbuss — CHANGES_REQUESTED** [review]

**Comment by marvinbuss** [comment]
@buchananwp Great job. Many thanks for the details and additions. This is great! 👍 

Could you please comment on the issues above?

**Review by will-iamalpine — COMMENTED** [review]
reset workspace.json and removed autogit artifacts

**Review by marvinbuss — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -51,31 +71,57 @@ This will generate the following JSON output:
...
+   "name": "your-workspace-name",
+   #same resource group as above
+    "resource_group": "your-resource-group",
+   #this stays the same
+    "create_workspace": true
+}
+```
+If you want the Action to create a new workspace in that resource group, 
+pick a name for your new workspace, and assign it to the `name` parameter. 
+You can also delete the `name` parameter, if you want the action to use the defau
```
Comment (marvinbuss) [inline README.md]: Typo: "defau" --> default

File: README.md [hunk 5]
```diff
@@ -51,31 +71,57 @@ This will generate the following JSON output:
...
+This is necessary so that your GitHub Actions will connect to your desired AzureML workspace. Click on the link and make your edits:
+
+```sh
+{
+   #same value for your-workspace-name as above
+   "name": "your-workspace-name",
+   #same resource group as above
+    "resource_group": "your-resource-group",
+   #this stays the same
+    "create_workspace": true
```
Comment (marvinbuss) [inline README.md]: Please fix the indentation.

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please just fix the 2 typos and then I will merge. Thanks for the great wok!

File: README.md (hunk 4 above)
Comment (will-iamalpine) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (will-iamalpine) [reply README.md]: done

**Comment by will-iamalpine** [comment]
@marvinbuss please review

File: README.md [hunk 6]
```diff
@@ -15,25 +15,52 @@ This template can be used for easily setting up a data science or machine learni
...
+A service principal (SP) needs to be generated for authentication and accessing your Azure subscription. We suggest adding a service principal with contributor rights to a new resource group, or to the one where you have deployed your existing Azure Machine Learning workspace. 
+
+Navigate to the [Azure Portal](https://www.portal.azure.com/) to find the details of your resource group or workspace. Create a Machine Learning workspace if you haven't already. 
+
+**Note:** If you can't add a role assignment, you are not permissioned with the ability to create a service principal. Contact your IT admin to remedy this. *(workspace > 'Access Control(IAM)' > 'Add a role assignment' > **'Add'**)*. 
+
+<p align="center">
+  <img src="/docs/images/workspace-permissions.png" alt="Workspace Permissions" width="550"/>
```
Comment (awmatheson) [inline README.md:41]: For consistency, could we update the images to make them styled similar to the `docs/images/secrets.png` where the sensitive info is blurred and the callout is shown with a red rectangle.

File: README.md [hunk 7]
```diff
@@ -15,25 +15,52 @@ This template can be used for easily setting up a data science or machine learni
...
+), in the section titled 'Service Principal Authentication'.
+
+**You will need the following to create your SP in the AZ CLI:**
+
+- **{service-principal-name}** = pick a new name for your service principal (e.g.workspacename_SPauth)
+- **{subscription-id}** = *Workspace > Overview > subscription ID*
+- **{resource-group}**= *Workspace > Overview > resource group*
+
+<p align="center">
+  <img src="/docs/images/workspace-overview.png" alt="Workspace Overview" width="550"/>
```
Comment (awmatheson) [inline README.md:56]: For consistency, could we update the images to make them styled similar to the `docs/images/secrets.png` where the sensitive info is blurred and the callout is shown with a red rectangle.

**Review by awmatheson — CHANGES_REQUESTED** [review]
The changes are great! We should merge this in once the images are changed and it's brought up to date with master.

**Pull request closed without merging**